src/context/AppProvider/AppContext.tsx

```diff
@@ -365,6 +375,48 @@ export const AppProvider = ({ children }: { children: React.ReactNode }) => {
     dispatch(marketApi.endpoints.findByAssetId.initiate(routeAssetId))
   }, [dispatch, routeAssetId])
 
+  useEffect(() => {
+    if (!isPortfolioLoaded || !wallet || !supportsETH(wallet)) return
+    // getting fox-eth lp token data
+    const {
+      getFoxEthLpGeneralData,
+      getFoxEthLpWalletData,
+      getFoxFarmingContractGeneralData,
+      getFoxFarmingContractWalletData,
+    } = foxEthApi.endpoints
+    dispatch(getFoxEthLpGeneralData.initiate())
+    const chainAdapterManager = getChainAdapterManager()
+    const adapter = chainAdapterManager.get(
+      ethChainId,
+    ) as ChainAdapter<KnownChainIds.EthereumMainnet>
+    if (foxEthLpMarketData.price === '0') return
+    ;(async () => {
+      const ethWalletAddress = await adapter.getAddress({ wallet })
+      // getting fox-eth lp token balances
+      // TODO: remove this condition when flags were removed
+      if (foxLpEnabled || foxFarmingEnabled)
+        dispatch(getFoxEthLpWalletData.initiate({ ethWalletAddress }))
+      farmOpportunities.forEach(opportunity => {
+        const { contractAddress } = opportunity
+        // getting fox farm contract data
+        dispatch(
+          getFoxFarmingContractGeneralData.initiate({
+            contractAddress,
+          }),
+        )
+        // getting fox farm contract balances
+        // TODO: remove this condition when flags were removed
+        if (foxFarmingEnabled)
+          dispatch(
+            getFoxFarmingContractWalletData.initiate({
+              contractAddress,
+              ethWalletAddress,
+            }),
+          )
+      })
+    })()
+  }, [dispatch, foxEthLpMarketData, foxFarmingEnabled, foxLpEnabled, isPortfolioLoaded, wallet])
+
```

Comment:
This is bringing FOX-ETH specific logic into the already fragile `<AppContext />`, this isn't the place for this.

From a high-level, what `<AppContext />` is is just a hook with many app-level dependencies and `useEffect()` calls, coupled with an effective single line of render guard:

https://github.com/shapeshift/web/blob/ccbfd841dee0c9294054879cb7b8c98c66ceb879/src/context/AppProvider/AppContext.tsx#L370

I get where you're coming from with this in wanting to handle things at app-"context" (we should probably revamp that terminology as well, since this isn't related to the context API) level, within a single source-of-truth component and avoiding multiple hooks instances, but we simply can't make this live here.

At the very least, this should be abstracted into its own hook and consumed here instead.

Going forward, I believe `<AppContext />` could be broken down and decoupled from the render logic, as the guts of this is really just a giant hook.